### PR TITLE
Fix Kazuha burst, Sara burst and Moonglow scaling

### LINKED ISSUE #413: Sara's burst does its damage at the same time its cooldown begins
Sara's burst should calculate and display its damage when her burst cooldown begins 50 frames after cast, but according to the sim, it happens 86 frames after the cooldown begins. This results in a 136 frame difference between casting her burst and doing damage, which is inaccurate. I can provide video evidence if needed.

### PR DESCRIPTION
**Pull request by shizukayuki**
- Cleanup Sara A4/C1 code
- Resolve #413

### CODE REVIEW

File: internal/characters/sara/abil.go [hunk 1]
```diff
@@ -191,74 +179,61 @@ func (c *char) Burst(p map[string]int) (int, int) {
...
-					// c.QueueDmg(&x, f+20+(50*(waveN+1)))
-					c.c1(f + 20 + (50 * (waveN + 1)))
-				}
-			}
-			if waveAttackProc == 1 {
-				c.attackBuff(f + 20 + (50 * (waveN + 1)))
-				c.c1(f + 20 + (50 * (waveN + 1)))
+	c1cb := make([]core.AttackCBFunc, 0, 1)
+	if c.Base.Cons >= 1 {
+		c1cb = append(c1cb, func(a core.AttackCB) { c.c1() })
```
[the remark this answers is not in the document]
Comment (shizukayuki) [reply internal/characters/sara/abil.go:204]: oh right that `make` call isn't needed.

File: internal/weapons/catalyst/moonglow/moonglow.go [hunk 2]
```diff
@@ -23,20 +23,19 @@ func weapon(char core.Character, c *core.Core, r int, param map[string]int) stri
...
-	nabuff := 0.0005 + float64(r)*0.0005
-	matk := make([]float64, core.EndStatType)
-	char.AddPreDamageMod(core.PreDamageMod{
-		Key: "moonglow-na-bonus",
-		Amount: func(atk *core.AttackEvent, t core.Target) ([]float64, bool) {
-			if atk.Info.AttackTag != core.AttackTagNormal {
-				return nil, false
-			}
+	nabuff := 0.005 + float64(r)*0.005
+	c.Events.Subscribe(core.OnAttackWillLand, func(args ...interface{}) bool {
```
[the remark this answers is not in the document]
Comment (shizukayuki) [reply internal/weapons/catalyst/moonglow/moonglow.go:27]: it is flat damage, redhorn has a similar description. i did verify it in-game with the currently running trial.

[the remark this answers is not in the document]
Comment (srliao) [reply internal/weapons/catalyst/moonglow/moonglow.go:27]: donut is a trap... why mhy :dies:

File: internal/characters/sara/abil.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (shizukayuki) [reply internal/characters/sara/abil.go]: i'm not sure if passing a nil function will be break the callbacks or not.

**Pull request merged**